python/packages/autogen-ext/src/autogen_ext/code_executors/jupyter/_jupyter_code_executor.py

```diff
@@ -0,0 +1,185 @@
+import asyncio
+import base64
+import json
+import re
+import sys
+import uuid
+from dataclasses import dataclass
+from pathlib import Path
+from types import TracebackType
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+from autogen_core import CancellationToken
+from autogen_core.code_executor import CodeBlock, CodeExecutor, CodeResult
+from nbclient import NotebookClient
+from nbformat import NotebookNode
+from nbformat import v4 as nbformat
+
+from .._common import silence_pip
+
+
+@dataclass
+class JupyterCodeResult(CodeResult):
+    """A code result class for Jupyter code executor."""
+
+    output_files: list[Path]
+
+
+class JupyterCodeExecutor(CodeExecutor):
+    def __init__(
+        self,
+        kernel_name: str = "python3",
+        timeout: int = 60,
+        output_dir: Path = Path("."),
+    ):
+        """A code executor class that executes code statefully using nbclient.
+
+        Args:
```

Comment:
Can you show a running example using the code executor directly. And perhaps another one wrapped inside the PythonCodeExecutionTool and used by an agent.

Reply:
There are test that show that. Or do you mean in the documentation? I think it would be nice to get some help from your for the documentation. At least for generating the skeleton. I think you have a better vision on where and how this fits into the documentation. For instance the previous documentation had a single code executor documentation, separated by local vs. docker I think. But you could separate commandline vs jupyter instead. Or have docs separated...
So maybed add some initial docs to this and I can fill it in with some more details / examples, where appropriate?

Reply:
I meant API docs -- the doc string to the classs. I will see if I can add a few examples directly to this PR later.